src/tool/cppwinrt/cppwinrt/code_writers.h

```diff
@@ -1337,33 +1337,49 @@ namespace xlang
         {
             w.write(R"(        void get() const
         {
-            blocking_suspend(static_cast<Windows::Foundation::IAsyncAction const&>(static_cast<D const&>(*this)));
-            GetResults();
-        })");
+            wait_get(static_cast<Windows::Foundation::IAsyncAction const&>(static_cast<D const&>(*this)));
+        }
+        auto wait_for(Windows::Foundation::TimeSpan const& timeout) const
+        {
+            return impl::wait_for(static_cast<Windows::Foundation::IAsyncAction const&>(static_cast<D const&>(*this)), timeout);
```

Comment:
>impl:: [](start = 19, length = 6)

needed for disambiguation with current wait_for?  (not used on wait_get)

Reply:
Right. `wait_for` calls `impl::wait_for` but `get` calls `impl::wait_get` so no confusion. I didn't want to call it `impl::get` and rather had the handful of impl functions start with `wait_xxx` for clarity.